coil-base/src/main/java/coil/fetch/HttpUrlFetcher.kt

```diff
@@ -16,40 +14,17 @@ import okhttp3.HttpUrl
 import okhttp3.Request
 import okhttp3.ResponseBody
 
-internal class HttpUriFetcher(callFactory: Call.Factory) : HttpFetcher<Uri>(callFactory) {
-
-    override fun handles(data: Uri) = data.scheme == "http" || data.scheme == "https"
-
-    override fun key(data: Uri) = data.toString()
-
-    override fun Uri.toHttpUrl(): HttpUrl = HttpUrl.get(toString())
-}
-
-internal class HttpUrlFetcher(callFactory: Call.Factory) : HttpFetcher<HttpUrl>(callFactory) {
+internal class HttpUrlFetcher(private val callFactory: Call.Factory) : Fetcher<HttpUrl> {
```

Comment:
Removed this optimization and re-added `HttpUriMapper` since mappers are now called on a background thread.